inc/utils.php

```diff
@@ -183,14 +187,26 @@ function _uploadcare_get_js_cfg() {
         }
     }
 
-    return array(
+    $baseParams = array(
         'public_key' => get_option('uploadcare_public'),
         'original' => get_option('uploadcare_original') ? "true" : "false",
         'multiple' => get_option('uploadcare_multiupload') ? "true" : "false",
         'previewStep' => $previewStep,
         'effects' => implode(',', $effects),
         'ajaxurl' => admin_url('admin-ajax.php'),
         'tabs' => $tabs,
-        'cdnBase' => 'https://' . get_option('uploadcare_cdn_base', 'ucarecdn.com')
+        'cdnBase' => 'https://' . get_option('uploadcare_cdn_base', 'ucarecdn.com'),
     );
+
+    $api = uploadcare_api();
+    $secureSignature = $api->widget->getSecureSignature();
+
+    if (!is_null($secureSignature)) {
+        return array_merge($baseParams, array(
+            'secureSignature' => $secureSignature->getSignature(),
+            'secureExpire' => $secureSignature->getExpire(),
+        ));
+    }
```

Comment:
should we put it into `get_option('uploadcare_lifetime') > 0` condition?

Reply:
yes, it would be good even add this check to uploadcare php, that lifetime is > 0.